File: 2014/Python/R-and-Python-Reproducing-R-functions-in-Python/Code.py
import numpy as np
import sys

def stcenter(x):
    
    if len(x.shape) is 1:
        sys.exit("ERROR: data must be a two dimensional array")
    elif x.shape[0] is 0:
        sys.exit("ERROR: data must have one or more observations for each location")
    elif x.shape[1] is 1:
        sys.exit("ERROR: data must have at least two locations")
    
    matrix = np.zeros(shape = x.shape)
    
    for j in range(x.shape[1]):
        matrix[:, j] = x[:, j] - np.mean(x[:, j])
        
    return matrix

File: 2014/Python/R-and-Python-Reproducing-R-functions-in-Python/Code.R
stcenter <- function(data){
  if(nrow(data) == 0L)
    stop("data must have one or more observations")
  if(ncol(data) == 1L)
    stop("data must have at least 2 locations")
  row <- dim(data)[1L]
  col <- dim(data)[2L]
  mat <- matrix(NA, nrow = row, ncol = col)
  for(j in 1L:col){
    mat[, j] <- data[, j] - mean(data[, j])      
  }  
  return(as.data.frame(mat))
}
